src/raftstore/store/store.rs

```diff
@@ -430,6 +437,12 @@ impl<T: Transport, C: PdClient> Store<T, C> {
                     }
                 }
 
+                if is_leader {
+                    // Notify pd quickly to let it update the region meta.
+                    self.heartbeat_pd(self.region_peers.get(&region_id).unwrap());
```

Comment:
Use a variable to store the returned peer at L430, then you can reuse the result.